src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/_params.py

```diff
@@ -13,6 +13,8 @@
 
 name_arg_type = CLIArgumentType(metavar='NAME', configured_default='botname', id_part='Name')
 
+supported_languages = ['Csharp', 'Node']
```

Comment:
Why is this not an enum in your SDK (which is another way of saying, why isn't this specified in Swagger)?

Reply:
Because this is not information that flows in the swagger, but higher level information. As you may know, we offer the basic CRUD operations for bot such as Create / Update / delete / Show but also some other operations that are outside of the basic crud such as Download and Publish for downloading and publishing bots. The languages are a construct of download and publish, but crud operations don't need to know the language generally.

The other scenario where languages are known is when we create a website for the bot (bot.kind == webapp). In which case we use the language to decide which bot template to publish to the website, but again it does not interact with our arm api.

Hope that makes sense, I'm happy to elaborate further.